tests/background_workload_test.py

```diff
@@ -0,0 +1,176 @@
+      spec.vms = [vm0, vm1]
+      path = os.path.join(os.path.dirname(__file__), 'data', 'ping.out')
+      outfile = open(path, 'r')
+      pingstdout = outfile.read()
+      for vm in spec.vms:
+        vm.RemoteCommand.side_effect = [(pingstdout, ' ')]
+
+      pkb.DoPreparePhase(ping_benchmark, NAME, spec, timer)
+      self.assertEquals(spec.Prepare.call_count, 1)
+
+      pkb.DoRunPhase(ping_benchmark, NAME, spec, collector, timer)
+      self.assertEquals(spec.StopBackgroundWorkload.call_count, 1)
+      self.assertEquals(spec.StartBackgroundWorkload.call_count, 1)
+
+  def testBackgroundWorkloadSpec(self):
+    """ Check that the benchmark spec calls the prepare, stop, and start
+    methods on the vms """
+
+    with mock_flags.PatchFlags() as mocked_flags:
+      mocked_flags.os_type = 'debian'
+      mocked_flags.machine_type = None
+      mocked_flags.cloud = 'GCP'
+      mocked_flags.background_cpu_threads = 1
+      collector = mock.MagicMock()
+      config = configs.LoadConfig(ping_benchmark.BENCHMARK_CONFIG, {}, NAME)
+      spec = benchmark_spec.BenchmarkSpec(config, NAME, UID)
+      vm0 = mock.MagicMock()
+      vm1 = mock.MagicMock()
+      spec.ConstructVirtualMachines()
+      spec.vms = [vm0, vm1]
+      timer = timing_util.IntervalTimer()
+      pkb.DoPreparePhase(ping_benchmark, NAME, spec, timer)
+      for vm in spec.vms:
+        self.assertEquals(vm.PrepareBackgroundWorkload.call_count, 1)
+
+      path = os.path.join(os.path.dirname(__file__), 'data', 'ping.out')
+      outfile = open(path, 'r')
+      pingstdout = outfile.read()
+
+      for vm in spec.vms:
+        vm.RemoteCommand.side_effect = [(pingstdout, ' ')]
+
+      pkb.DoRunPhase(ping_benchmark, NAME, spec, collector, timer)
+      for vm in spec.vms:
+        self.assertEquals(vm.StartBackgroundWorkload.call_count, 1)
+        self.assertEquals(vm.StopBackgroundWorkload.call_count, 1)
+
+  def testBackgroundWorkloadVM(self):
+    """ Check that the vm background workload calls work """
+    with mock_flags.PatchFlags() as mocked_flags:
+      mocked_flags.os_type = 'debian'
+      mocked_flags.machine_type = None
+      mocked_flags.cloud = 'GCP'
+      mocked_flags.background_cpu_threads = 1
+      config = configs.LoadConfig(ping_benchmark.BENCHMARK_CONFIG, {}, NAME)
+      spec = benchmark_spec.BenchmarkSpec(config, NAME, UID)
+      spec.ConstructVirtualMachines()
+      [vm0, vm1] = spec.vms
+      with patch.object(vm0.__class__, 'RemoteCommand'):
+        with patch.object(vm0.__class__, 'Install'):
+          spec.Prepare()
+          self.assertTrue(vm0.Install.call_count, 1)
+          # check that start called remote command.
+          spec.StartBackgroundWorkload()
+          self.assertTrue(vm0.RemoteCommand.call_count, 1)
+          # check that stop called remote command.
+          spec.StopBackgroundWorkload()
+          self.assertTrue(vm0.RemoteCommand.call_count, 2)
+
+  def testBackgroundWorkloadVanillaConfig(self):
```

Comment:
Unlike the previous test, this one doesn't appear to be verifying whether the functions were called.